### PR TITLE
vdirsyncer: fix verify option type

### PR DESCRIPTION
**Pull request by BagMan9**
### Description

Vdirsyncer options for calendar/contacts accounts specify the "verify" option as types.bool. However, this is not how vdirsyncer treats the option - it expects a path to certificate to verify against.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://nix-community.github.io/home-manager/#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

#### Maintainer CC
@teto

### CODE REVIEW

File: modules/programs/vdirsyncer-accounts.nix [hunk 1]
```diff
@@ -93,9 +93,9 @@ in {
     };
 
     verify = mkOption {
-      type = types.nullOr types.bool;
+      type = types.nullOr types.path;
       default = null;
-      description = "Verify SSL certificate.";
```
Comment (teto) [inline modules/programs/vdirsyncer-accounts.nix:98]: Could you add the example from http://vdirsyncer.pimutils.org/en/stable/ssl-tutorial.html#custom-root-cas ?

**Comment by teto** [comment]
LGTM, see http://vdirsyncer.pimutils.org/en/stable/ssl-tutorial.html#custom-root-cas

File: modules/programs/vdirsyncer-accounts.nix (hunk 1 above)
Comment (BagMan9) [reply modules/programs/vdirsyncer-accounts.nix:98]: Added.

File: modules/programs/vdirsyncer.nix [hunk 2]
```diff
@@ -92,7 +92,7 @@ let
     else if (n == "passwordPrompt") then
       ''password.fetch = ["prompt", "${v}"]''
     else if (n == "verify") then
-      "verify = ${if v then "true" else "false"}"
+      ''verify = "${v}"''
```
Comment (teto) [inline modules/programs/vdirsyncer.nix:95]: out of scope but this function is weird and unmaintainable :/ we should have a more transparent mapping !

**Pull request merged**